instrumentation/active_support/lib/opentelemetry/instrumentation/active_support/span_subscriber.rb

```diff
@@ -137,11 +138,20 @@ def sanitized_value(value)
           end
         end
 
+        def span_name(name)
```

Comment:
I see what you meant now. Duping the string will now result in a separate string allocation and an additional case statement per subscription call. The difference in use between the regexp matcher and string based subscriber is only for span names right?

I think maybe it would be best in this case then to use the span name found on the payload and then keep the name instance variable as more as a indexing value that we can use to unsubscribe registrations or inspect the object to know what it's subscribed to but not use it for span naming. Does that help?

Reply:
Thank you for your review @arielvalentin!

> I think maybe it would be best in this case then to use the span name found on the payload and then keep the name instance variable as more as a indexing value that we can use to unsubscribe registrations or inspect the object to know what it's subscribed to but not use it for span naming. Does that help?

It does help. It's also a behavioural change, meaning this test will break:

https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/7bda424d9c1ab4c26be9cb9697558e7fcd1a3d3e/instrumentation/active_support/test/opentelemetry/instrumentation/active_support/span_subscriber_test.rb#L38-L41

Which is the behaviour I was trying to preserve there. In this case, would it make sense to just remove that test?

Reply:
Thank you! I've updated that on b8c0fca1a6c44d1800b67639a975eb36a5d117c3. Can you please check if this is what you were looking for?